src/Serilog.Settings.Configuration/Settings/Configuration/ConfigurationReader.cs

```diff
@@ -154,27 +154,39 @@ Assembly[] LoadConfigurationAssemblies()
                 }
             }
 
-            foreach (var library in _dependencyContext.RuntimeLibraries)
+            foreach (var assemblyName in GetSerilogConfigurationAssemblies())
             {
-                if (library.Name != null && library.Name.ToLowerInvariant().Contains("serilog"))
-                {
-                    var assumedName = new AssemblyName(library.Name);
-                    var assumed = Assembly.Load(assumedName);
-                    if (assumed != null && !assemblies.ContainsKey(assumed.FullName))
-                        assemblies.Add(assumed.FullName, assumed);
-
-                    foreach (var assemblyRef in library.Assemblies)
-                    {
-                        var assembly = Assembly.Load(assemblyRef.Name);
-                        if (assembly != null && !assemblies.ContainsKey(assembly.FullName))
-                            assemblies.Add(assembly.FullName, assembly);
-                    }
-                }
+                var assumedName = new AssemblyName(assemblyName);
+                var assumed = Assembly.Load(assumedName);
+                if (assumed != null && !assemblies.ContainsKey(assumed.FullName))
+                    assemblies.Add(assumed.FullName, assumed);
             }
 
             return assemblies.Values.ToArray();
         }
 
+        string[] GetSerilogConfigurationAssemblies()
+        {
+            var query = Enumerable.Empty<string>();
+            var filter = new Func<string, bool>(name => name != null && name.ToLowerInvariant().Contains("serilog"));
+
+            if (_dependencyContext != null)
+            {
+                query = from lib in _dependencyContext.RuntimeLibraries where filter(lib.Name) select lib.Name;
+            }
+            else
+            {
+#if APPDOMAIN
+                query = from outputAssemblyPath in System.IO.Directory.GetFiles(AppDomain.CurrentDomain.BaseDirectory, "*.dll", System.IO.SearchOption.AllDirectories)
+                        let assemblyFileName = System.IO.Path.GetFileNameWithoutExtension(outputAssemblyPath)
+                        where filter(assemblyFileName)
+                        select Assembly.ReflectionOnlyLoadFrom(outputAssemblyPath).GetName().FullName;
```

Comment:
Just wanted to make GetSerilogConfigurationAssemblies() pure with best intent to take only metadata from the assembly and not load it with it references (the actual load in LoadConfigurationAssemblies()).
But found more appropriate AssemblyName.GetAssemblyName(path). Fixed.